server/base/src/main/java/org/apache/accumulo/server/util/MasterMetadataUtil.java

```diff
@@ -197,13 +181,6 @@ public static void replaceDatafiles(ServerContext context, KeyExtent extent,
     if (compactionId != null)
       tablet.putCompactionId(compactionId);
 
-    TServerInstance self = getTServerInstance(address, zooLock);
-    tablet.putLocation(self, LocationType.LAST);
-
-    // remove the old location
-    if (lastLocation != null && !lastLocation.equals(self))
-      tablet.deleteLocation(lastLocation, LocationType.LAST);
-
```

Comment:
Why is it the case that these location entries no longer need to be updated here? Where is the replacement code?

Reply:
Same, as above.